### PR TITLE
Fix Grayscale Image Shape

### PR DESCRIPTION
**Pull request by Torrencem**
When using grayscale images, making the shape [None, width, height, 1] will allow you to use it directly with an input_layer, since it's a 4-D tensor, as apposed to [None, width, height]. Color images are 4-D by default ([None, width, height, 3])

I ran into this problem when trying to do the following:

`...`
`X, Y = image_preloader(dataset_file, image_shape=(75, 90), mode='file', categorical_labels=True,` `normalize=True, grayscale=True)`

`# Building deep neural network`
`network = tflearn.input_data(shape=[None, 90, 75, 1]) #Can't be [None, 90,75], because must be 4-D` `tensor`
`...`

Without the change, tensorflow raises an exception that the tensor for the images needs to be 4-D as well. This is a pretty simple fix for that

### CODE REVIEW

File: tflearn/data_utils.py [hunk 1]
```diff
@@ -832,6 +832,8 @@ def preload(self, path, image_shape, normalize=True, grayscale=False):
         if grayscale:
             img = convert_color(img, 'L')
         img = pil_to_nparray(img)
+        if grayscale:
+            img = np.reshape(img, (*img.shape, 1))
```
Comment (aymericdamien) [inline tflearn/data_utils.py]: *img.shape through an exception.

**Comment by Torrencem** [comment]
Should be fixed now, (img.shape*, 1) doesn't work on 2.x I think, so changed it to tuple addition
